### PR TITLE
What's On - Replace Abandoned 3rd Party Share Feature with Simplified Facebook and URL Copy Sharing; Code Refactoring

### PR DESCRIPTION
**Pull request by Horib**
### Why this Pull Request Exists:

This Pull Request was prompted by the discontinuation of a 3rd party share feature that our platform relied on. To address this, we've introduced a simpler, in-house sharing solution, allowing users to share content via Facebook or by copying the URL. This ensures our users can continue to easily share content while we maintain control over the functionality. Concurrently, we've undertaken code refactoring to enhance the overall quality and maintainability of our codebase.

**Monday Task:** https://visitfaroeislands.monday.com/boards/4732496989/pulses/5077183818

### Changes Made:

- Introduced a Facebook share button and a URL copy button, providing users with straightforward, alternative sharing options after the discontinuation of the 3rd party share feature. These changes aim to preserve and even enhance user engagement and content dissemination.

- Eliminated the deprecated 3rd party share functionalities, preventing potential user confusion or reliance on unsupported services.

- Conducted extensive code refactoring, including the implementation of the `SubmoduleLinkHandler` for more dynamic and maintainable path handling across various components (hiking, places, tours, posters, submit), and the renaming of variables (`absoluteWhatsonViews` to `viewPath`, `absoluteWhatsonPath` to `basePath`, and `subWhatson` to `isSubmodule`) for increased clarity.

- Performed a general clean-up of the code to bolster its readability, maintainability, and adherence to best coding practices.

### Test Guide for the Reviewer:

**Facebook Share Button:**
- Navigate to a page with shareable content.
- Locate and click the new Facebook share button.
- Ensure that the correct link is being shared on Facebook, the content appears appropriate, and the post visibility settings are functional.

**URL Copy Button:**
- Go to a page with shareable content.
- Find and click the URL copy button.
- Verify that the correct URL is copied to your clipboard.
- Paste the URL elsewhere to ensure it directs to the correct content.

**Refactoring and Deprecated Functionality Removal:**
- Paths and Variable Renaming:
- Navigate through the sections impacted by the refactoring.
- Confirm functionality remains intact, especially link navigations.
- Deprecated Share Features:
- Ensure that remnants of the old 3rd party share features are completely removed from the user interface.

**General Code Review:**
- Review the code changes to ensure consistency, optimal performance, and the absence of redundant or obsolete code.

### CODE REVIEW

File: views/js/utils/favouritesLikedCount.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+export function favouritesLikedCount(type, array) {
+    let numbers = [3, 7, 2, 15, 4, 9, 21, 14];
```
Comment (HemmingAskodan) [inline views/js/utils/favouritesLikedCount.js:2]: What is this??!?

File: views/style/_event.scss [hunk 2]
```diff
@@ -27,37 +27,37 @@
...
-          border: 1px solid rgba($color: #a2a2a2, $alpha: 0.2);
-          border-radius: 100%;
-          color: black;
-
-          @media ($phone) {
-            display: none;
-          }
-        }
-      }
+      // & > .social {
```
Comment (HemmingAskodan) [inline views/style/_event.scss:30]: Remove out commented code.

File: views/style/_hike.scss [hunk 3]
```diff
@@ -51,21 +51,21 @@
         // padding-top: 40px;
         padding-bottom: 0;
       }
-      & > .social {
```
Comment (HemmingAskodan) [inline views/style/_hike.scss:54]: Remove out commented code

File: views/style/_place.scss [hunk 4]
```diff
@@ -29,36 +29,36 @@
...
-          justify-content: center;
-          border: 1px solid rgba($color: #a2a2a2, $alpha: 0.2);
-          border-radius: 100%;
-
-          @media ($phone) {
-            display: none;
-          }
-        }
-      }
+      // & > .social {
```
Comment (HemmingAskodan) [inline views/style/_place.scss:32]: Remove out commented code

File: views/style/_tour.scss [hunk 5]
```diff
@@ -27,23 +27,23 @@
...
-          > i {
-            position: absolute;
-            top: 50%;
-            left: 50%;
-            transform: translate(-50%, -50%);
-            font-size: 18px;
-          }
-        }
-      }
+      // & > .social {
```
Comment (HemmingAskodan) [inline views/style/_tour.scss:30]: Remove out commented code

**Review by HemmingAskodan — APPROVED** [review]
Small cleanup can be done. I have commented on the changes that can be done.

File: views/js/utils/favouritesLikedCount.js (hunk 1 above)
Comment (Horib) [reply views/js/utils/favouritesLikedCount.js:2]: No idea

**Pull request merged**